crates/networking/rpc/rogue/transaction.rs

```
@@ -0,0 +1,231 @@
+                value: U256::zero(),
+                data: self.data.clone(),
+                access_list: Vec::new(),
+                ..Default::default()
+            })
+        };
+
+        let mut generic = match tx.to_transaction() {
+            ethrex_common::types::Transaction::EIP1559Transaction(tx) => {
+                GenericTransaction::from(tx)
+            }
+            ethrex_common::types::Transaction::EIP7702Transaction(tx) => {
+                GenericTransaction::from(tx)
+            }
+            _ => unreachable!("This should never happen, we are creating the tx"),
+        };
+        generic.gas = None;
+
+        let estimate_gas_request = EstimateGasRequest {
+            transaction: generic,
+            block: None,
+        }
+        .handle(context.clone())?;
+        let gas_limit = u64::from_str_radix(
+            estimate_gas_request
+                .as_str()
+                .unwrap_or("0x0")
+                .strip_prefix("0x")
+                .unwrap(),
+            16,
+        )
+        .unwrap();
+        if gas_limit == 0 || gas_limit > GAS_LIMIT_HARD_LIMIT {
+            return Err(RpcErr::InvalidRogueMessage("tx too expensive".to_string()));
+        }
+        match tx {
+            SendRawTransactionRequest::EIP7702(ref mut tx) => {
+                tx.gas_limit = gas_limit;
+                tx.max_fee_per_gas = max_fee_per_gas;
+                tx.max_priority_fee_per_gas = max_priority_fee_per_gas;
+                tx.nonce = nonce;
+                tx.sign_inplace(&sponsor_pk);
+            }
+            SendRawTransactionRequest::EIP1559(ref mut tx) => {
+                tx.gas_limit = gas_limit;
+                tx.max_fee_per_gas = max_fee_per_gas;
+                tx.max_priority_fee_per_gas = max_priority_fee_per_gas;
+                tx.nonce = nonce;
+                tx.sign_inplace(&sponsor_pk);
+            }
+            _ => unreachable!("This should never happen, we are creating the tx"),
```

Let's return an error here instead, you never know what will happen if we ever introduce a bug reach this path.

---

It will panic. It's cleaner to error out, but it's not UB. That would be with the `unsafe` version, `unreachable_unchecked`.

---

changed it [here](https://github.com/lambdaclass/ethrex/pull/2214/commits/83c3c831a387bdf6c78e3c339dc1c33d81db222d) and forgot the return so [here](https://github.com/lambdaclass/ethrex/pull/2214/commits/cf4f5fc7e8b232060d30cdd827328a9d299e7b75) is the fix for that